simtools/simtel/simtel_events.py

```diff
@@ -1,194 +1,253 @@
+                    if energy < energyRange[0] or energy > energyRange[1]:
+                        continue
 
-            fig = plt.figure(figsize=(8, 6))
-            ax = plt.gca()
-            ax.set_title(hist['title'])
+                    if r_core > coreMax:
+                        continue
 
-            if hist['n_bins_y'] > 0:
-                # 2D histogram
+                    selectedEvents.append(event)
+        return selectedEvents
 
-                xlim = _get_ax_lim(hist, axis=0)
-                ylim = _get_ax_lim(hist, axis=1)
+    @u.quantity_input(coreMax=u.m)
+    def countSimulatedEvents(self, energyRange=None, coreMax=None):
+        '''
+        Count (or calculate) number of simulated events within a certain energy range and \
+        core radius, nased on the simulated power law.
 
-                if np.sum(hist['data']) == 0:
-                    ax.text(
-                        0.5,
-                        0.5,
-                        'EMPTY',
-                        horizontalalignment='center',
-                        verticalalignment='center',
-                        transform=ax.transAxes
-                    )
-                    continue
+        Parameters
+        ----------
+        energyRange: Tuple (len 2)
+            Max and min energy of energy range, e.g. energyRange=(100 * u.GeV, 10 * u.TeV)
+        coreMax: astropy.Quantity (distance)
+            Maximum core radius for selecting showers, e.g. coreMax=1000 * u.m
+
+        Returns
+        -------
+        int
+            Number of simulated events.
+        '''
+        energyRange = self._validateEnergyRange(energyRange)
+        coreMax = self._validateCoreMax(coreMax)
 
-                x_bins = _get_bins(hist, axis=0)
-                y_bins = _get_bins(hist, axis=1)
+        # energy factor
+        def integral(erange):
+            power = self._mcHeader['spectral_index'] + 1
+            return math.pow(erange[0], power) - math.pow(erange[1], power)
 
-                ax.pcolormesh(x_bins, y_bins, hist['data'])
-                ax.set_xlim(xlim)
-                ax.set_ylim(ylim)
+        energy_factor = integral(energyRange) / integral(self._mcHeader['E_range'])
 
-            else:
-                # 1D histogram
+        # core factor
+        core_factor = math.pow(coreMax, 2) / math.pow(self._mcHeader['core_range'][1], 2)
 
-                xlim = _get_ax_lim(hist, axis=0)
+        return self._mcHeader['n_events'] * energy_factor * core_factor
 
-                if np.sum(hist['data']) == 0:
-                    ax.text(
-                        0.5,
-                        0.5,
-                        'EMPTY',
-                        horizontalalignment='center',
-                        verticalalignment='center',
-                        transform=ax.transAxes
-                    )
-                    continue
+    def _validateEnergyRange(self, energyRange):
```

Comment:
You are missing documentation for this function.

Also, I would not expect that a function called "validate" would return the full energy range in case I did not supply one.

Reply:
I added some docstrings.

Well, it returns the default in case energyRange=None is provided.
I agree it is not the perfect name but I think it makes sense in this context. Note that it is a private/internal function.

Reply:
Yea, the name isn't a big deal in this case since it's an internal function, as long as the docstring makes it clear.